### ISSUE TITLE
/api/v2/user/auth returning 406 (!) when attempting to log in

### ISSUE THREAD

**Issue by Hixie** [seamonkeysocial/twitarr#285]
When trying to log in with an invalid account, using RainbowMonkey:
```
>>> GET http://69.62.137.54:7001/api/v2/user/auth?username=hixie&password=testtest
<<< (!!) 406 from api/v2/user/auth?username=hixie&password=testtest
```
There's no payload in the response (Content-Length: 0).

According to https://github.com/seamonkeysocial/twitarr/blob/master/rest_documentation.md it should return either a payload (presumably with 200 OK though I guess we don't document that) or 401 Unauthorized with a payload.

I checked the Web site and it's using POST (though not with JSON, which I guess violates the API description) and is getting back 204 No Content with a cookie set (which also violates the API description), and notably is not actually reporting the error either so presumably the Web site doesn't expect this situation either.

Logging in with a valid account works fine both on Web and on Rainbow Monkey.

cc @hendricksond

**Comment by Hixie** [comment]
Changing passwords also seems broken when you provide bad data, at least if my experience changing kvort's password is anything to go by.

**Comment by hendricksond** [comment]
Fixed by #292
